homeassistant/components/azuredns/__init__.py

```diff
@@ -0,0 +1,176 @@
+               resourcegroupname +
+               '/providers/Microsoft.Network/dnsZones/' +
+               domain +
+               '/A/' +
+               host +
+               '?api-version=2018-05-01')
+
+    async_get_clientsession(hass)
+
+    result = await _update_azuredns(resource, tenant,
+                                    clientid, clientsecret,
+                                    authority_url, api_url, timeout, ttl)
+
+    if not result:
+        _LOGGER.error("Failed to update Azure DNS record.")
+        return False
+
+    async def update_domain_interval():
+        """Update the Azure DNS entry."""
+        await _update_azuredns(resource, tenant,
+                               clientid, clientsecret, authority_url,
+                               api_url, timeout, ttl)
+
+    async_track_time_interval(hass, update_domain_interval, INTERVAL)
+    return result
+
+
+async def _update_azuredns(resource, tenant,
+                           clientid, clientsecret, authority_url,
+                           api_url, timeout, ttl):
+    """Update the Azure DNS Record with the external IP address."""
+    import adal
+    import aiohttp
+
+    params = {
+        'resource': resource,
+        'tenant': tenant,
+        'clientid': clientid,
+        'clientsecret': clientsecret,
+        'api_url': api_url,
+    }
+
+    # Acquire the Azure AD token.
+    context = adal.AuthenticationContext(
+        authority_url, validate_authority=['tenant'] != 'adfs',
+    )
+
+    token = context.acquire_token_with_client_credentials(
+        params['resource'],
+        params['clientid'],
+        params['clientsecret'])
+
+    access_token = token.get('accessToken')
+
+    if not access_token:
+        _LOGGER.error("Failed to acquire Azure AD Access Token.")
+    else:
+        _LOGGER.debug("Azure AD App Token was acquired.")
+
+    # Get the external IP address of the Home Assistant instance.
+    aiotimeout = aiohttp.ClientTimeout(total=timeout)
+
+    async with aiohttp.ClientSession(timeout=aiotimeout) as aiosession:
+        async with aiosession.get('https://api.ipify.org') as aioresp:
+            ipv4address = (await aioresp.text())
+
+    if not ipv4address:
+        _LOGGER.error("Failed to get external IP address from ipify API")
+    else:
+        _LOGGER.debug("External IP address is: %s", ipv4address)
+
+    # Create a PATCH request with the Azure Resource Manager REST API.
+    headers = {
+        "Authorization": "Bearer " + access_token,
+        "Content-Type": "application/json"
+    }
+    url = params['api_url']
+    payload = {
+        "properties": {
+            "metadata": {
+                "Last_changed_by_Home_Assistant": str(datetime.now())
+            },
+            "TTL": ttl,
+            "ARecords": [
+                {
+                    "ipv4Address": ipv4address
+                }
+            ]
+        }
+    }
+
+    try:
+        result = requests.patch(url, json=payload,
```

Comment:
See previous comment about `requests` and `aiohttp`.

Reply:
Your comment on that was: "I think that we should go with the aiohttp_client and not using requests additionally."

I will work on replacing the "requests" request with aiohttp_client.

Reply:
In the meantime we try to get rid of `requests` (#23685).

Reply:
Thanks for the clarification.